src/plugins/orangehrmPerformancePlugin/Api/PerformanceReviewFinalEvaluationAPI.php

```diff
@@ -0,0 +1,257 @@
+                    [
+                        PerformanceReview::class,
+                        (new InAccessibleEntityIdOption())
+                            ->setRolesToExclude(['ESS'])
+                    ]
+                )
+            ),
+            new ParamRule(
+                self::PARAMETER_COMPLETE,
+                new Rule(Rules::BOOL_TYPE)
+            ),
+            ...$this->getFinalEvaluationBodyParamRules()
+        );
+    }
+
+    /**
+     * @return ParamRule[]
+     */
+    private function getFinalEvaluationBodyParamRules(): array
+    {
+        $paramRules = [
+            new ParamRule(
+                self::PARAMETER_FINAL_RATING,
+                new Rule(Rules::FLOAT_TYPE),
+                new Rule(
+                    Rules::BETWEEN,
+                    [self::PARAM_RULE_FINAL_RATING_MIN_VALUE, self::PARAM_RULE_FINAL_RATING_MAX_VALUE]
+                )
+            ),
+            new ParamRule(
+                self::PARAMETER_FINAL_COMMENT,
+                new Rule(Rules::STRING_TYPE)
+            ),
+            new ParamRule(
+                self::PARAMETER_COMPLETED_DATE,
+                new Rule(Rules::API_DATE)
+            ),
+        ];
+
+        $complete = $this->getRequestParams()->getBoolean(
+            RequestParams::PARAM_TYPE_BODY,
+            self::PARAMETER_COMPLETE
+        );
+
+        if (!$complete) {
+            return array_map(function ($rule) {
+                return $this->getValidationDecorator()
+                    ->notRequiredParamRule($rule);
+            }, $paramRules);
+        }
+        return $paramRules;
+    }
+
+    /**
+     * @param PerformanceReview $performanceReview
+     */
+    private function setFinalEvaluation(PerformanceReview $performanceReview): void
+    {
+        $performanceReview->setFinalRate(
+            $this->getRequestParams()->getFloatOrNull(
+                RequestParams::PARAM_TYPE_BODY,
+                self::PARAMETER_FINAL_RATING,
+            )
+        );
+        $performanceReview->setFinalComment(
+            $this->getRequestParams()->getStringOrNull(
+                RequestParams::PARAM_TYPE_BODY,
+                self::PARAMETER_FINAL_COMMENT
+            )
+        );
+        $performanceReview->setCompletedDate(
+            $this->getRequestParams()->getDateTimeOrNull(
+                RequestParams::PARAM_TYPE_BODY,
+                self::PARAMETER_COMPLETED_DATE
+            )
+        );
+        $performanceReview->setStatusId(
+            PerformanceReview::STATUS_IN_PROGRESS
+        );
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function getOne(): EndpointResult
+    {
+        $reviewId = $this->getRequestParams()->getInt(
+            RequestParams::PARAM_TYPE_ATTRIBUTE,
+            self::PARAMETER_REVIEW_ID
+        );
+
+        $review = $this->getPerformanceReviewService()
+            ->getPerformanceReviewDao()
+            ->getPerformanceReviewById($reviewId);
```

Comment:
Should we check for review status Id in here?

Reply:
```
$this->throwBadRequestExceptionIfStatusInArray(
            $review->getStatusId(),
            [PerformanceReview::STATUS_INACTIVE, PerformanceReview::STATUS_COMPLETED]
        );
```
throwing BadRequestException if status matches inactive or completed. For get one operation, only checking for inactive state